common/src/main/java/io/netty/util/HashingStrategy.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util;
+
+/**
+ * Abstraction for hash code generation and equality comparison.
+ */
+public interface HashingStrategy<T> {
+    /**
+     * Generate a hash code for {@code obj}.
+     */
+    int hashCode(T obj);
+
+    /**
+     * Check if {@code a} equals {@code b}.
+     */
+    boolean equals(T a, T b);
+
+    /**
+     * A {@link HashingStrategy} which delegates to java's {@link java.lang.Object#hashCode()}
+     * and {@link java.lang.Object#equals(Object)}.
+     */
+    @SuppressWarnings("rawtypes")
+    HashingStrategy JAVA_HASHER = new HashingStrategy() {
+        @Override
+        public int hashCode(Object obj) {
+            return obj.hashCode();
+        }
+
+        @Override
+        public boolean equals(Object a, Object b) {
+            return a.equals(b);
```

Comment:
I was wondering if we need to compare nulls using `Objects.equals(a, b)` so that NPE is not raised.